src/common/history.c

```diff
@@ -570,31 +584,56 @@ static int _history_copy_and_paste_on_image_merge(int32_t imgid, int32_t dest_im
 
       if(hist)
       {
-        // merge the entry
-        dt_history_merge_module_into_history(dev_dest, dev_src, hist->module, &modules_used, FALSE);
+        double old_iop_order = hist->module->iop_order;
+
+        if (iop_order_version_src != iop_order_version_dest)
+          hist->module->iop_order = dt_ioppr_get_iop_order(dest_iop_list, hist->module->op) + (double)hist->module->multi_priority / 100.0f;
+
+        if (!dt_iop_is_hidden(hist->module))
+        { if (DT_IOP_ORDER_INFO)
+          fprintf(stderr,"\n  module %20s, order %9.5f->%9.5f, multiprio %i",
+                hist->module->op, old_iop_order, hist->module->iop_order, hist->module->multi_priority);
+
+          // merge the entry
+          dt_history_merge_module_into_history(dev_dest, dev_src, hist->module, &modules_used, FALSE);
+        }
       }
 
       l = g_list_previous(l);
     }
   }
   else
   {
+    if (DT_IOP_ORDER_INFO) fprintf(stderr," all modules");
     // we will copy all modules
     GList *modules_src = g_list_first(dev_src->iop);
     while(modules_src)
     {
       dt_iop_module_t *mod_src = (dt_iop_module_t *)(modules_src->data);
 
-      // but only if module is in history in source image
-      if(_search_history_by_module(dev_src, mod_src) != NULL)
+      // copy from history only if
+      if((_search_history_by_module(dev_src, mod_src) != NULL) &&  // module is in history of source image
+         !(mod_src->default_enabled && !mod_src->enabled) &&       // it's not a disabled module normally enabled by default
+         !dt_iop_is_hidden(mod_src) &&                             // it's not a hidden module
+         !memcmp(mod_src->params, mod_src->default_params, mod_src->params_size)
```

Comment:
This means that a module with default param will be excluded and so will not override dest module with non default params?

Reply:
Confirmed by testing:
- select two images reset history
- go to image A, set exposure to +2EV
- go to image B set exposure to default (click reset settings)
- go to lighttable, select history on B (ctrl+c)
- paste on A, exposure is not changed and set to +2EV

To me this is not the expected behavior.

Reply:
Note that copying default params is the only way to reset a module on a set of pictures otherwise the only solution is to go to each image individually and disable the module.

Reply:
Yeah, that's wrong. memcmp check is only relevant for default_enabled. Correct if should look like that:
```
if((_search_history_by_module(dev_src, mod_src) != NULL) && // module is in history of source image
!(mod_src->default_enabled && mod_src->enabled && !memcmp(mod_src->params, mod_src->default_params, mod_src->params_size) && // it's not a enabled by default module with unmodified settings
!dt_iop_is_hidden(mod_src)) // it's not a hidden module
```

Reply:
Of course you are right :-(